File: ruby/basics/fibonacci.rb
#!/usr/bin/env ruby
# Print the first 20 Fibonacci numbers from a lazy Enumerator.

fibonacci = Enumerator.new do |yielder|
  current = 0
  following = 1
  loop do
    yielder << current
    current, following = following, current + following
  end
end

puts fibonacci.take(20).join(' ')

File: python/basics/fibonacci.py
#!/usr/bin/env python3
# Print the first 20 Fibonacci numbers from a generator.

from itertools import islice


def fibonacci():
    current, following = 0, 1
    while True:
        yield current
        current, following = following, current + following


print(" ".join(str(value) for value in islice(fibonacci(), 20)))
